protocol/contracts/templegold/TempleTeleporter.sol

```diff
@@ -0,0 +1,108 @@
+pragma solidity ^0.8.20;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Temple (templegold/TempleTeleporter.sol)
+
+
+import { CommonEventsAndErrors } from "contracts/common/CommonEventsAndErrors.sol";
+import { ITempleERC20Token } from "contracts/interfaces/core/ITempleERC20Token.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { OApp } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oapp/OApp.sol";
+import { MessagingReceipt, MessagingFee } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oapp/OAppSender.sol";
+import { Origin } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oapp/interfaces/IOAppReceiver.sol";
+import { ITempleTeleporter } from "contracts/interfaces/templegold/ITempleTeleporter.sol";
+import { OFTMsgCodec } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oft/libs/OFTMsgCodec.sol";
+
+contract TempleTeleporter is ITempleTeleporter, OApp {
+    using OFTMsgCodec for address;
+
+    /// @notice Temple token
+    ITempleERC20Token public immutable override temple;
+
+    constructor(
+        address _executor,
+        address _temple,
+        address _endpoint
+    ) Ownable(_executor) OApp(_endpoint, _executor){
+        temple = ITempleERC20Token(_temple);
+    }
+
+    /**
+     * @notice Teleport temple tokens cross chain
+     * @dev Temple tokens are burned from source chain and minted on destination chain
+     * @param dstEid Destination chain id
+     * @param to Recipient
+     * @param amount Amount of tokens
+     * @param options LZ extra options
+     */
+    function teleport(
+        uint32 dstEid,
```

Comment:
This will be reverted when checked at the `MessageSendLibrary` of the endpoint. There won't be a configured library for same chain.
```
function getSendLibrary(address _sender, uint32 _dstEid) public view returns (address lib) {
        lib = sendLibrary[_sender][_dstEid];
        if (lib == DEFAULT_LIB) {
            lib = defaultSendLibrary[_dstEid];
            if (lib == address(0x0)) revert Errors.LZ_DefaultSendLibUnavailable();
        }
    }
```